docs/documentation/development/guidelines.md

```diff
@@ -26,13 +36,22 @@ Note that this list also serves as a checklist for code reviews on pull requests
 1. Packages that contain classes that are not meant to be used by other bundles should have "internal" in their package name.
 1. We are using [null annotations](https://wiki.eclipse.org/JDT_Core/Null_Analysis) from the Eclipse JDT project. Therefore every bundle should have an **optional** `Import-Package` dependency to `org.eclipse.jdt.annotation`.
 Classes should be annotated by `@NonNullByDefault` and return types, parameter types, generic types etc. are annotated with `@Nullable` only.
+For fields that store injected services, i.e. they will never be `null` in an OSGI framework, we annotate them with
+
+```java
+@NonNullByDefault({})
+private MyService injectedService;
+```
+
+to skip the `null` evaluation for these fields.
+Fields within `ThingHandler` classes that are initialized within the `initialize()` method may also be annotated like this, because the framework ensures that they will never be `null`.
```

Comment:
> because the framework ensures that they will never be `null`

This is not correct. It only (tries to) ensure that no methods are called before initialize() has been called. It's the duty of the handler implementation to ensure that such fields are set in initialize(). Note that I have seen many situations where fields are NOT set if the initialization fails and the Thing is set to OFFLINE - so this can actually be a tricky part.